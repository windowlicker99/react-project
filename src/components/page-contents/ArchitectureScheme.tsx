import React, { FC, MouseEventHandler, useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  ConnectorModel,
  DataBinding,
  DiagramComponent,
  Inject,
  NodeModel,
  SnapConstraints,
  SnapSettingsModel,
} from '@syncfusion/ej2-react-diagrams';
import { TooltipComponent } from '@syncfusion/ej2-react-popups';
import { createNodes } from '@utils/createNodes';
import { createConnectors } from '@utils/createConnectors';
import { calculateSchemeOffsetAndScale } from '@utils/calculateSchemeOffsetAndScale';
import {
  IArchitectureDomain,
  IDiagramClickEventArgs,
  ISelectionChangeArgs,
  IDiagramTooltip,
  IElementAddInfo,
} from '@interfaces/ArchitectureScheme/interfaces';
import { setCurrentElement } from '@store/slices/elementSlice';
import { selectedItemsSettings, tooltipAnimationParams } from '@constants/ArchitectureScheme/parameters';
import { createDiagramTooltip } from '@utils/createDiagramTooltip';
import { selectCurrentVehicle } from '@store/selectors/vehicle';
import { createGateway } from '@utils/createGateway';
import { PageMock } from './PageMock';
import { getAddedNodes } from '@/utils/getAddedNodes';
import { EElementPropName } from '@/interfaces/ArchitectureScheme/enums';
import { TDiagramEventElement } from '@/interfaces/ArchitectureScheme/types';
import { IECUasGateway, IGatewayResponse } from '@/interfaces/interfaces';

interface IArchitectureSchemeProps {
  onECUCreateClick: () => void;
  schemeData: IArchitectureDomain[];
  gatewayInfo: IGatewayResponse;
}

const cloneDeep = require('lodash.clonedeep');

export const ArchitectureScheme: FC<IArchitectureSchemeProps> = ({ schemeData, gatewayInfo, onECUCreateClick }) => {
  const dispatch = useDispatch();
  const currentVehicle = useSelector(selectCurrentVehicle);
  const [gatewayNode, setGatewayNode] = useState<NodeModel>(null);
  const [nodes, setNodes] = useState<NodeModel[]>([]);
  const [connectors, setConnectors] = useState<ConnectorModel[]>([]);
  const [tooltip, setTooltip] = useState<IDiagramTooltip>({ text: '' });
  const [selectedConnectors, setSelectedConnectors] = useState([]);
  const clonedConnectors = cloneDeep(connectors);
  const [addedNodes, setAddedNodes] = useState<NodeModel[]>([]);
  let diagramInstance: DiagramComponent = null;

  if (!currentVehicle) {
    return <PageMock />;
  }

  const changeSelectedConnectorStyle = (element: TDiagramEventElement, width: number) => {
    if (!element?.addInfo || element.addInfo.relative) {
      return;
    }
    connectors.forEach((connector, i) => {
      const domainId = `domain${element.addInfo.element?.id}_to`;
      if (connector.id.includes(domainId)) {
        clonedConnectors[i].style.strokeWidth = width;
      }
    });
  };

  const getNodeHTML = (element: TDiagramEventElement) =>
    document.querySelector(`#${element.id}_content_html_element > div > div`);

  const changeSelectedStyle = (element: TDiagramEventElement, isNew: boolean) => {
    if (element?.data === EElementPropName.connector) {
      const width = isNew ? 2 : 1;
      changeSelectedConnectorStyle(element, width);
    }
    if (element?.data === EElementPropName.node) {
      const node = getNodeHTML(element);
      if (!node) {
        return;
      }
      if (isNew) {
        node.classList.add('selected');
        return;
      }
      node.classList.remove('selected');
    }
  };

  const manageStylesOfSelectedItems = (args: ISelectionChangeArgs): void => {
    const {
      newValue: [newValue],
      oldValue: [oldValue],
    } = args;

    changeSelectedStyle(newValue, true);
    changeSelectedStyle(oldValue, false);

    // update connectors
    setConnectors(clonedConnectors);
  };

  const changeColorConnection = (args: ISelectionChangeArgs) => {
    const {
      newValue: [element],
    } = args;
    const relativeConnectors: number[] = [];

    connectors.forEach((connector, index) => {
      const addInfo = connector.addInfo as IElementAddInfo;

      if (
        (connector.sourceID === element?.id && addInfo?.relative) ||
        (element?.id === connector.targetID && addInfo?.relative)
      ) {
        relativeConnectors.push(index);
      }
    });

    relativeConnectors.forEach((i) => {
      clonedConnectors[i].style = {
        strokeColor:
          clonedConnectors[i].style.strokeColor === '#BEBEBE' ? clonedConnectors[i].addInfo?.relative.color : '#BEBEBE',
      };
    });

    selectedConnectors.forEach((i) => {
      clonedConnectors[i].style = {
        strokeColor:
          clonedConnectors[i].style.strokeColor === '#BEBEBE' ? clonedConnectors[i].addInfo?.relative.color : '#BEBEBE',
      };
    });

    setSelectedConnectors(relativeConnectors);

    setConnectors(clonedConnectors);
  };

  const getGatewayNode = async () => {
    if (!gatewayInfo) {
      setGatewayNode(null);
      return;
    }
    const updateGatewayNode = createGateway(gatewayInfo);

    setGatewayNode(updateGatewayNode);
  };

  const getNodes = () => {
    const schemeNodes = createNodes(schemeData, gatewayNode);
    if (nodes.length) {
      const added = getAddedNodes(nodes, schemeNodes);
      setAddedNodes(added);
    }
    setNodes(gatewayNode ? [gatewayNode, ...schemeNodes] : schemeNodes);
    const gateway = gatewayInfo?.gateway as IECUasGateway;
    const schemeConnectors = createConnectors(schemeData, gatewayNode?.height, gateway?.color);
    setConnectors(schemeConnectors);
  };

  useEffect(() => {
    getGatewayNode();
  }, [gatewayInfo]);

  useEffect(() => {
    if (schemeData) {
      getNodes();
    }
  }, [schemeData, gatewayNode]);

  const snapSettings: SnapSettingsModel = {
    constraints: SnapConstraints.None,
  };

  const onShowElementDetails = (args: ISelectionChangeArgs) => {
    const {
      newValue: [element],
    } = args;
    if (element?.addInfo) {
      const addInfo = element.addInfo as IElementAddInfo;
      if (addInfo.element) {
        dispatch(setCurrentElement(addInfo.element));
      }
    } else {
      dispatch(setCurrentElement(null));
    }
  };

  const onSelectElement = (args: ISelectionChangeArgs) => {
    manageStylesOfSelectedItems(args);
    if (!args.newValue?.length) {
      dispatch(setCurrentElement(null));
    } else {
      onShowElementDetails(args);
      changeColorConnection(args);
    }
  };

  const onDiagramMisclick = ({ actualObject }: IDiagramClickEventArgs) => {
    if (!actualObject) {
      diagramInstance.select([]);
    }
  };

  const onMouseOver = ({ actualObject }: IDiagramClickEventArgs) => {
    if (actualObject?.addInfo) {
      const { addInfo } = actualObject;

      if (addInfo.tooltip) {
        setTooltip(addInfo.tooltip);
      }
    }
  };

  const onMouseLeave = () => {
    setTooltip({ text: '' });
  };

  const onOpenModal: MouseEventHandler = (e) => {
    const { target } = e;
    if (!(target instanceof HTMLElement)) {
      return;
    }
    if (target.dataset.createEcu) {
      onECUCreateClick();
    }
  };

  useEffect(() => {
    // to update nodes in the diagram correctly
    if (diagramInstance) {
      if (addedNodes.length) {
        diagramInstance.clearSelection(); // remove selection from add button
        diagramInstance.select(addedNodes); // show details of new element
      }
      nodes.forEach((node) => {
        diagramInstance.add(node);
      });
    }
  }, [nodes]);

  useEffect(() => {
    if (diagramInstance) {
      // preserves the interactivity of the connectors
      diagramInstance.connectors = [...connectors];

      // fixes the issue of disappearing connectors
      connectors.forEach((connector) => {
        diagramInstance.add(connector);
      });
    }
  }, [connectors]);

  useEffect(
    () => () => {
      diagramInstance = null;
    },
    []
  );

  if (nodes?.length) {
    return (
      <TooltipComponent
        isSticky
        closeDelay={300}
        content={createDiagramTooltip(tooltip)}
        mouseTrail
        showTipPointer
        position="BottomCenter"
        offsetY={10}
        animation={tooltipAnimationParams}
      >
        <DiagramComponent
          ref={(diagram) => {
            diagramInstance = diagram;
          }}
          selectedItems={selectedItemsSettings}
          id="diagram"
          width="100%"
          height="100%"
          snapSettings={snapSettings}
          scrollSettings={calculateSchemeOffsetAndScale(schemeData?.length)}
          mouseOver={onMouseOver}
          mouseLeave={onMouseLeave}
          selectionChange={onSelectElement}
          click={onDiagramMisclick}
          onClick={onOpenModal}
        >
          <Inject services={[DataBinding]} />
        </DiagramComponent>
      </TooltipComponent>
    );
  }

  return null;
};
