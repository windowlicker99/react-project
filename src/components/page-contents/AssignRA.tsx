import React, { FC, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { useFormik } from 'formik';
import { ModalCenter } from '@components/modules/ModalCenter';
import { AssignTicketModalContent } from '@components/modal-content/AssignTicketModalContent';
import { selectCurrentVehicle } from '@store/selectors/vehicle';
import { getUsers } from '@utils/getUsers';
import { convertUsersToOptions } from '@utils/converters';
import { getElement } from '@utils/getElement';
import { createTicket } from '@utils/createTicket';
import { ICurrentElement, IOption } from '@interfaces/interfaces';
import { TElement } from '@interfaces/types';
import { EAsideActualElement, ETicketsTypes } from '@interfaces/enums';

interface IAssignRA {
  element: ICurrentElement;
  isOpen: boolean;
  onClose: () => void;
}

export const AssignRA: FC<IAssignRA> = ({ element, isOpen, onClose }) => {
  const defaultOption = { label: 'Not Assigned', value: 'NaN' };
  const [currentElement, setCurrentElement] = useState<TElement>(null);
  const [usersOptions, setUsersOptions] = useState<IOption[]>([defaultOption]);

  const vehicleId = useSelector(selectCurrentVehicle);

  const getUsersOptions = async () => {
    const users = await getUsers();
    const options = convertUsersToOptions(users);
    setUsersOptions([defaultOption, ...options]);
  };

  const getCurrentElement = async () => {
    const newElement = await getElement(element);
    setCurrentElement(newElement);
  };

  useEffect(() => {
    if (element) {
      getCurrentElement();
    }
  }, [element]);

  useEffect(() => {
    getUsersOptions();
  }, []);

  const formik = useFormik({
    initialValues: { user: defaultOption, message: '' },
    onSubmit: async ({ user: { value: userId }, message }, { resetForm }) => {
      try {
        await createTicket({
          name: `Risk Assessment for ${currentElement?.name}`,
          type: ETicketsTypes.riskAssessment,
          elementName: currentElement.name,
          elementType: EAsideActualElement[element.type],
          elementId: element.id,
          userId,
          vehicleId,
          message,
        });
      } catch (error) {
        console.log(error);
      } finally {
        resetForm();
        onClose();
      }
    },
  });

  const handleSelectChange = (name: string, value: IOption) => formik.setFieldValue(name, value);

  const onApplyRAModal = () => {
    formik.submitForm();
  };

  const onCloseRAModal = () => {
    formik.resetForm();
    onClose();
  };

  return (
    <ModalCenter
      title="Assign Ticket"
      subtitle={`Risk Assessment for ${currentElement?.name}`}
      isOpen={isOpen}
      disabledApplyBtn={!formik.values.user || formik.values.user.value === defaultOption.value}
      onApply={onApplyRAModal}
      onClose={onCloseRAModal}
    >
      <AssignTicketModalContent
        values={formik.values}
        options={usersOptions}
        onChange={formik.handleChange}
        onSelectChange={handleSelectChange}
      />
    </ModalCenter>
  );
};
