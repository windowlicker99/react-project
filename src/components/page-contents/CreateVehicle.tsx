import React, { useState, FC } from 'react';
import { Form, Formik, FormikProps } from 'formik';
import { ModalAside } from '@components/modules/ModalAside';
import { ModalFooter } from '@components/modal-content/modal-aside/ModalFooter';
import { ModalHeader } from '@components/modal-content/modal-aside/ModalHeader';
import { ResultContent } from '@components/modal-content/ResultContent';
import { VehicleCreateInfoContent } from '@components/modal-content/VehicleCreateInfoContent';
import { VehicleCreateDatesContent } from '@components/modal-content/VehicleCreateDatesContent';
import { getFieldsValues } from '@utils/getFieldsValues';
import { gatewayInfo, vehicleCreationDates, vehicleCreationElements } from '@constants/formsElements';
import { createVehicle } from '@utils/createVehicle';
import { convertVehicleFormValuesToPostData } from '@utils/converters';

import { vehicleBasicInfoValidationScheme, vehicleMilestoneDatesValidationScheme } from '@utils/validations';
import { IGatewayFormValues, IVehicleFormValues } from '@/interfaces/interfaces';

interface ICreateVehicle {
  isOpen: boolean;
  onClose: () => void;
}

type TPages = 1 | 2;

export const CreateVehicle: FC<ICreateVehicle> = ({ isOpen, onClose }) => {
  const [step, setStep] = useState<TPages>(1);
  const initialValues = {
    ...getFieldsValues(Object.values(vehicleCreationElements)),
    gateway: getFieldsValues(Object.values(gatewayInfo)) as IGatewayFormValues,
    ...getFieldsValues(Object.values(vehicleCreationDates)),
  };

  const [successfullyAdded, setSuccessfullyAdded] = useState<string>('');

  const onSubmit = async (values: IVehicleFormValues) => {
    try {
      const vehicle = convertVehicleFormValuesToPostData(values);
      await createVehicle(vehicle);
      setSuccessfullyAdded(`${values.brand?.label} | ${values.model} | ${values.platform} was created successfully!`);
    } catch (error) {
      console.log(error);
    }
  };

  const onConfirmClick = () => {
    setSuccessfullyAdded('');
    setStep(1);
    onClose();
  };

  const onPrevPageClick = () => setStep(1);
  const onNextPageClick = () => setStep(2);

  const onCloseModal = () => {
    setStep(1);
    onClose();
  };

  const pages = {
    1: {
      component: <VehicleCreateInfoContent />,
      header: {
        title: 'Create Vehicle',
        subtitle: 'Basic Information',
      },
      footer: {
        applyBtn: 'Next',
      },
      validationScheme: vehicleBasicInfoValidationScheme,
      onSubmit: onNextPageClick,
    },
    2: {
      component: <VehicleCreateDatesContent />,
      header: {
        title: 'Create Vehicle',
        subtitle: 'Milestone Dates',
      },
      footer: {
        applyBtn: 'Complete Creation',
        onBack: onPrevPageClick,
      },
      validationScheme: vehicleMilestoneDatesValidationScheme,
      onSubmit,
    },
  };

  return successfullyAdded ? (
    <ModalAside isOpen={isOpen}>
      <ResultContent title="Success" content={successfullyAdded} onConfirm={onConfirmClick} />
    </ModalAside>
  ) : (
    <Formik
      initialValues={initialValues}
      validationSchema={pages[step].validationScheme}
      onSubmit={pages[step].onSubmit}
    >
      {({ submitForm, resetForm, isValid, setTouched }: FormikProps<Partial<IVehicleFormValues>>) => (
        <ModalAside
          isOpen={isOpen}
          header={
            <ModalHeader
              step={step}
              maxStep={2}
              {...pages[step].header}
              onClose={() => {
                resetForm();
                onCloseModal();
              }}
            />
          }
          footer={
            <ModalFooter
              {...pages[step].footer}
              onApply={() => {
                submitForm();
                if (isValid) {
                  setTouched({});
                }
              }}
            />
          }
        >
          <Form> {pages[step].component}</Form>
        </ModalAside>
      )}
    </Formik>
  );
};
