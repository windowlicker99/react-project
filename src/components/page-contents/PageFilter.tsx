import React, { FC } from 'react';
import classNames from 'classnames';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { Btn } from '@components/common/form-controls/Btn';
import { IFilter } from '@interfaces/interfaces';

interface IPageFilter {
  filters?: Array<IFilter>;
  isOpen?: boolean;
  close?: () => void;
}

export const PageFilter: FC<IPageFilter> = ({ filters = [], isOpen = false, close }) => {
  const onApplyFilters = () => {
    // there should be some apply logic
    close();
  };

  return (
    <div className={classNames('page-filter', isOpen ? 'show' : '')}>
      <div className="page-filter-inner">
        {filters.map(({ label, name }) => (
          <div className="page-filter-item" key={name}>
            <SelectComponent label={label} placeholder={label} />
          </div>
        ))}
      </div>
      <div className="page-filter-btn">
        <Btn btnStatus="btn-xs-secondary mr-15" btnIcon="icon-close" onClick={close} />
        <Btn btnStatus="btn-lg-outline" btnText="Apply" onClick={onApplyFilters} />
      </div>
    </div>
  );
};

// QUESTION <div className="page-filter"> <!-- add className show -->
