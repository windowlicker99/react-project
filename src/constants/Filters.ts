export const defaultFilters = [
  {
    name: 'brand',
    label: 'Brand',
  },
  {
    name: 'model',
    label: 'Model',
  },
  {
    name: 'platform',
    label: 'Platform',
  },
  {
    name: 'phase',
    label: 'Phase',
  },
  {
    name: 'regulation',
    label: 'Regulation',
  },
];

export const domainFilters = [
  {
    name: 'model',
    label: 'Model',
  },
  {
    name: 'platform',
    label: 'Platform',
  },
];
