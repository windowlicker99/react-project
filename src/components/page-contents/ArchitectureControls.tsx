import React, { FC, useEffect, useState } from 'react';
import { Btn } from '@components/common/form-controls/Btn';
import { Badge } from '@components/common/Badge';
import { Legend } from '@components/common/Legend';
import { PageFilter } from '@components/page-contents/PageFilter';
import { CircleChart } from '@components/common/CircleChart';
import { domainFilters } from '@constants/Filters';
import { IVehicle } from '@interfaces/interfaces';
import { EBadgeTypes, ERiskLevels } from '@interfaces/enums';
import { initialVehicle } from '@constants/initialData';

interface IArchitectureControls {
  vehicle: IVehicle;
  onOpenDomainModalClick: () => void;
  onOpenHistoryView: () => void;
}

export const ArchitectureControls: FC<IArchitectureControls> = ({
  vehicle,
  onOpenDomainModalClick,
  onOpenHistoryView,
}) => {
  const [displayedVehicle, setDisplayedVehicle] = useState<IVehicle>(initialVehicle);
  const [isOpenFilters, setIsOpenFilters] = useState<boolean>(false);

  const onShowFiltersClick = () => {
    setIsOpenFilters(!isOpenFilters);
  };

  const refreshFilters = () => {
    // temporary solution. Will be removed with adding filters refresh logic
    console.log('refresh filters');
  };

  useEffect(() => {
    if (!vehicle) {
      setDisplayedVehicle(initialVehicle);
      return;
    }
    setDisplayedVehicle(vehicle);
  }, [vehicle]);

  return (
    <div className="box page-controls">
      <div className="page-controls-content">
        <div className="page-controls-item">
          <div className="mr-15">
            <Btn
              disabledBtn={!vehicle}
              btnStatus="btn-lg-primary"
              btnIcon="icon-plus"
              btnText="Create Domain"
              onClick={onOpenDomainModalClick}
            />
          </div>
          <div className="mr-15">
            <Btn btnStatus="btn-lg-outline" btnText="View History" onClick={onOpenHistoryView} />
          </div>
          <div className="mr-15">
            <Btn btnStatus="btn-xs-primary" btnIcon="icon-filter-blue" onClick={onShowFiltersClick} />
          </div>
          <div className="mr-15">
            <Btn btnStatus="btn-xs-secondary" btnIcon="icon-refresh" onClick={refreshFilters} />
          </div>
        </div>
        <div className="page-controls-item">
          <div className="page-controls-architecture">
            <div className="page-controls-data">
              {vehicle && (
                <div className="page-controls-info">
                  <span className="page-controls-name">{displayedVehicle.brand?.name}</span>
                  {displayedVehicle.model}
                </div>
              )}
            </div>
            <div className="page-controls-data">
              Risk Level
              <Badge
                badgeType={EBadgeTypes[displayedVehicle.riskLevel]}
                badgeText={ERiskLevels[displayedVehicle.riskLevel]}
              />
            </div>
            <div className="page-controls-data">
              Progress
              <div className="chart">
                <CircleChart percentage={displayedVehicle.raProgress} />
              </div>
            </div>
            <div className="page-controls-data">
              <Legend isDisabled={!vehicle} legendType="Eth" legendName="Ethernet" />
              <Legend isDisabled={!vehicle} legendType="CAN" legendName="CAN" />
              <Legend isDisabled={!vehicle} legendType="LIN" legendName="LIN" />
            </div>
          </div>
        </div>
      </div>
      <PageFilter filters={domainFilters} isOpen={isOpenFilters} close={onShowFiltersClick} />
    </div>
  );
};
