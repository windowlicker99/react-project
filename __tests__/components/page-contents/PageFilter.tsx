import React from 'react';
import { render } from '@testing-library/react';
import { PageFilter } from '@components/page-contents/PageFilter';
import { defaultFilters } from '@constants/Filters';

describe('PageFilter tests', () => {
  it('Should render', () => {
    const { container } = render(<PageFilter filters={defaultFilters} isOpen />);
    expect(container.getElementsByClassName('page-filter').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(<PageFilter filters={defaultFilters} isOpen />);
    expect(asFragment()).toMatchSnapshot();
  });
});
