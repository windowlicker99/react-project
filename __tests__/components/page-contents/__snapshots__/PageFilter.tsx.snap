// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFilter tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-filter show"
  >
    <div
      class="page-filter-inner"
    >
      <div
        class="page-filter-item"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="exampleFormControlInput1"
          >
            Brand
          </label>
          <div
            class="select-component css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="react-select__control css-1pgkwyd-control"
            >
              <div
                class="react-select__value-container css-g1d714-ValueContainer"
              >
                <div
                  class="react-select__placeholder css-oco7ot-placeholder"
                >
                  Brand
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-7-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <div
                    class="form-arrow icon-arrow-border-down"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-filter-item"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="exampleFormControlInput1"
          >
            Model
          </label>
          <div
            class="select-component css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="react-select__control css-1pgkwyd-control"
            >
              <div
                class="react-select__value-container css-g1d714-ValueContainer"
              >
                <div
                  class="react-select__placeholder css-oco7ot-placeholder"
                >
                  Model
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-8-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <div
                    class="form-arrow icon-arrow-border-down"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-filter-item"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="exampleFormControlInput1"
          >
            Platform
          </label>
          <div
            class="select-component css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="react-select__control css-1pgkwyd-control"
            >
              <div
                class="react-select__value-container css-g1d714-ValueContainer"
              >
                <div
                  class="react-select__placeholder css-oco7ot-placeholder"
                >
                  Platform
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-9-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <div
                    class="form-arrow icon-arrow-border-down"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-filter-item"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="exampleFormControlInput1"
          >
            Phase
          </label>
          <div
            class="select-component css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="react-select__control css-1pgkwyd-control"
            >
              <div
                class="react-select__value-container css-g1d714-ValueContainer"
              >
                <div
                  class="react-select__placeholder css-oco7ot-placeholder"
                >
                  Phase
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-10-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <div
                    class="form-arrow icon-arrow-border-down"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-filter-item"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="exampleFormControlInput1"
          >
            Regulation
          </label>
          <div
            class="select-component css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class="react-select__control css-1pgkwyd-control"
            >
              <div
                class="react-select__value-container css-g1d714-ValueContainer"
              >
                <div
                  class="react-select__placeholder css-oco7ot-placeholder"
                >
                  Regulation
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class="react-select__input"
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-11-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                >
                  <div
                    class="form-arrow icon-arrow-border-down"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="page-filter-btn"
    >
      <button
        class="btn btn-xs-secondary mr-15"
        type="button"
      >
        <span
          class="btn-icon icon-close"
        />
        <span
          class="btn-text"
        />
      </button>
      <button
        class="btn btn-lg-outline"
        type="button"
      >
        <span
          class="btn-icon "
        />
        <span
          class="btn-text"
        >
          Apply
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
